aea/aea_builder.py

```diff
@@ -1246,6 +1243,70 @@ def set_from_configuration(
                 skip_consistency_check=skip_consistency_check,
             )
 
+        skill_ids = [
+            ComponentId(ComponentType.SKILL, p_id)
+            for p_id in agent_configuration.skills
+        ]
+        skill_import_order = self._find_import_order(skill_ids, aea_project_path)
+        for skill_id in skill_import_order:
+            component_path = self._find_component_directory_from_component_id(
+                aea_project_path, skill_id
+            )
+            self.add_component(
+                skill_id.component_type,
+                component_path,
+                skip_consistency_check=skip_consistency_check,
+            )
+
+    def _find_import_order(self, skill_ids, aea_project_path):
+        """Find import order for skills.
+
+        We need to handle skills separately, since skills can depend on each other.
+        That is, we need to:
+        - load the skill configurations to find the import order
+        - detect if there are cycles
+        - import skills from the leaves of the dependency graph, by finding a topological ordering.
+        """
+        # the adjacency list for the dependency graph
+        depends_on: Dict[PublicId, Set[PublicId]] = defaultdict(set)
+        # the adjacency list for the inverse dependency graph
+        supports: Dict[PublicId, Set[PublicId]] = defaultdict(set)
+        # nodes with no incoming edges
+        roots = copy(skill_ids)
+
+        for skill_id in skill_ids:
+            component_path = self._find_component_directory_from_component_id(
+                aea_project_path, skill_id
+            )
+            configuration = cast(
+                SkillConfig,
+                ComponentConfiguration.load(
+                    skill_id.component_type, component_path, False
+                ),
+            )
+
+            if len(configuration.skills) != 0:
+                roots.remove(skill_id)
+            depends_on[skill_id].update(configuration.skills)
+            for dependency in configuration.skills:
+                supports[dependency].add(skill_id)
```

Comment:
this loop finds the dependencies relationships between skills. For that we need to read the configuration files of each skill.